Add more information for Android in connection security level (#1265)
home-assistant/companion.home-assistant: docs/getting_started/connection_security_level.mdx

title: "Connection security level"
id: "connection-security-level"
---

> If you use a plain text connection using HTTP instead of an encrypted connection using HTTPS, then will have to choose between two security options in the app if you want to use a unencrypted URL to connect to Home Assistant (such as your local IP address for local connection).

Reviewer: You added info at the start that duplicates what is already at the end of the sentence. Suggestion:

```suggestion
If you want to use a unencrypted URL to connect to Home Assistant (such as your local IP address for local connection), you will have to set the connection security level. You can choose between two security options in the app.
```

:::note
This option are mandatory to set while onboarding a new server. For users that 
didn't set this option yet it is going to be prompt on each start of the app
until a choice is made.
:::

## Option 1: Most secure (recommended)

This option ensures the app only connect to a unencrypted URL (for example, `http://homeassistant.local:8123`) when you are currently on your specified home network. When you are not on the specified home network, the app will display a screen blocking access and describing the requirements to connect. This also means that background activity will be blocked until a secure connection can be established.

You can specify your Wi-Fi SSID (for example, "GrandmaHome5G") to be considered your home network. To change the setting open the [Settings](https://my.home-assistant.io/redirect/config/) > **Companion app** > **[server name]** anytime you want.

> <details>

Reviewer: Two details containers looks quite busy. What about a single container "Additional options" which describes Android and macOS?

Author: I wanted to hide what is not relevant for a user when he is looking at the documentation
<img width="997" height="571" alt="image" src="https://github.com/user-attachments/assets/e50d81f8-4c86-46cd-a20b-c37d9f3a36ce" />

You feel that is takes too much space?

  <summary>**Android**</summary>

You can also use a VPN or Ethernet access as home network.
</details>

<details>
  <summary>**macOS**</summary>

You can define a "Hardware address" as home network, which is useful for wired connections.
</details>

To retrieve the information above, the app needs the location access permission from your device, since network information could be used for location positioning. We understand that this is not desirable for all users, but this restriction is imposed by Apple platforms and Android. Your location will never be used for anything besides retrieving the network information. You are always in control if you want to share your location to your local Home Assistant server.

:::warning Important
The location permission must be set to **Always** and **Precise** for iOS and **Allow all the time** for Android for the app to function correctly in the background.
:::

> ### Implication when not in secure state

Author: I applied your suggestion and added a new FAQ entry.

Choosing "Less secure" may expose your Home Assistant credentials if you connect from a public network. Make sure you trust all networks you connect to when using this option.
:::

### What happens when I'm not on my home network with "Most secure" enabled?

> The app will block access and display a screen explaining the situation. Background activity (widgets, sensors, notifications) will also pause until a secure connection can be established.

Reviewer: I just realized notifications will probably use FCM for most users so are unaffected in delivery. This only affects websocket notifications, or if your notifications result in actions that need to reach the server. That's too much complexity here so I propose simply removing it.

```suggestion
The app will block access and display a screen explaining the situation. Background activity (for example, widgets and sensors) will also pause until a secure connection can be established.
```

Choosing "Less secure" may expose your Home Assistant credentials if you connect from a public network. Make sure you trust all networks you connect to when using this option.
:::

### How do I set up my home network?

1. Grant location permission when prompted (required for the **Most secure** option).
2. Connect to your home Wi-Fi network.

> 3. Go to [Settings](https://my.home-assistant.io/redirect/config/) > **Companion app** > **[your server]** > **Internal URL**.

Reviewer: Shouldn't Settings still be bolded?

4. Add your home Wi-Fi SSID to the list.

You can add multiple SSIDs if you have more than one home network.

### Can I specify the Hardware Address (Ethernet) instead of just the SSID?

On macOS, yes. On iOS, this is not currently available due to platform limitations.
This is not an option on Android.

### Does this feature work when I'm connected through a VPN?

<details>
  <summary>**Android**</summary>

  You can select the VPN option while setting up your home network in [Settings](https://my.home-assistant.io/redirect/config/) > **Companion app** > **[your server]** > **Home network**.
  
  :::note
  It cannot detect on which VPN it connects only that it is connected to a VPN or not.

>   :::

Reviewer: ```suggestion
  You can select the **VPN connected** option as home network in [Settings](https://my.home-assistant.io/redirect/config/) > **Companion app** > **[your server]** > **Home network**.
  
  :::note
  This option only detects connection to *any* VPN. Connection to a specific VPN cannot be detected due to platform limitations.
  :::
```
